test/utils/componentInfo.js

```diff
@@ -0,0 +1,37 @@
+import _ from 'lodash'
+import path from 'path'
+
+const componentCtx = require.context(
+  '../../src/',
+  true,
+  /(addons|collections|elements|modules|views).*\.js$/
+)
+
+const componentInfo = _.map(componentCtx.keys(), key => {
+  const Component = componentCtx(key)
+  const { _meta, prototype } = Component
+
+  const constructorName = prototype.constructor.name
+  const filePath = key
+  const filename = path.basename(key)
+  const filenameWithoutExt = path.basename(key, '.js')
+  const subComponentName = _meta.parent ? _meta.name.replace(_meta.parent, '') : null
+  // HeaderH1 => sd-header-h1
+  const sdClass = 'sd-' + constructorName
```

Comment:
Again, totally agree.  Once we hit 1.0, let's profile and double back to make the biggest improvements.